pandas/tests/test_algos.py

```diff
@@ -639,7 +646,7 @@ def test_datetime64_dtype_array_returned(self):
             ]
         )
         result = algos.unique(dt_index)
-        tm.assert_numpy_array_equal(result, expected)
+        tm.assert_index_equal(result, to_datetime(expected))
```

Comment:
`expected` should be exactly what's expected and not need adjustment. Can you make the modification to the computation of `expected` itself.

Reply:
I see.

I kept the array `expected` because it is used again below. Now I rename the variable to make it more explicit.